tensorboard/uploader/formatters.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Helpers that format experiment metadata as strings."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import json
+
+EXPERIMENT_METADATA_URL_JSON_KEY = "url"
+ExperimentMetadataField = collections.namedtuple(
+    "ExperimentMetadataField",
+    ("json_key", "readable_name", "value", "formatter"),
```

Comment:
I was confused by the name collision of "formatter".  Can this one be "field_formatter" or "to_readable" or similar?

Also, it still feels a bit intertwined to pass a lambda that is specifically used only in the ReadableFormatter case.  I think you can remove the field entirely and automate this.  In `ReadableFormatter.format_experiment`, you could:

```
if not metadata_field.value and not typeof(metadata_field.value) == 'int':
  readable_value = "[No %s]".format(metadata_field.readable_name)
else:
  readable_value = str(metadata_field.value)
```

The typecheck there is a little iffy (because, as written, `None` will trigger the substitution even for fields that should be int, and a future float-valued field containing `0.0` would trigger too).  I guess `ExperimentMetadataField` could carry a Type enum, but that seems overwrought.  I think it'd be fine to gloss over all that for now (maybe with a code comment about the limitations).